new-nextjs-client/src/components/index-page/AddYearDropdown/AddYearDropdown.tsx

```diff
@@ -24,11 +26,33 @@ const getYearOptions = (): YearOption[] => {
 
 function AddYearDropdown() {
   const { appUser, updateAppUser } = useAppUser();
+  const [yearOptions, setYearOptions] = useState(getYearOptions());
+
+  const showToastBox = useAppToast();
 
   const onSelectChange = (newValue: YearOption) => {
-    updateAppUser((draft) => {
-      draft.addYear(newValue.value);
-    });
+    if (!appUser.years.includes(newValue.value)) {
+      updateAppUser((draft) => {
+        draft.addYear(newValue.value);
+        draft.sortYear();
+
+        showToastBox({
```

Comment:
call `showToastBox()` outside the `updateAppUser()`.

updateAppUser should only include code related to updating the state of AppUser